contrib/cmake/build3rdParty/build3rdParty.cpp

```diff
@@ -0,0 +1,273 @@
+#include <iostream>
+#include <filesystem>
+#include <vector>
+#include <map>
+#include <string>
+#include <fstream>
+
+using namespace std;
+
+
+#if (__cplusplus >= 201703L && (!defined(__GNUC__) || (__GNUC__*100+__GNUC_MINOR__) >= 800))
+#include <filesystem>
+namespace fs = std::filesystem;
+#define USE_FILESYSTEM
+#elif !defined(__MINGW32__) && !defined(__ANDROID__) && (!defined(__GNUC__) || (__GNUC__*100+__GNUC_MINOR__) >= 503)
+#define USE_FILESYSTEM
+#ifdef WIN32
+#include <filesystem>
+namespace fs = std::experimental::filesystem;
+#else
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+#endif
+#else
+//for mac
+#include <filesystem>
+namespace fs = std::__fs::filesystem;
+#endif
+
+
+
+bool build = false;
+bool setup = false;
+bool removeUnusedPorts = false;
+fs::path portsFile;
+fs::path sdkRootPath;
+string triplet;
+
+map<string, string> ports;
+
+fs::path initialDir = fs::current_path();
+fs::path vcpkgDir = fs::current_path() / "vcpkg";
+fs::path cloneDir = fs::current_path() / "vcpkg_clone";
+
+
+bool readCommandLine(int argc, char* argv[]);
+void execute(string command);
+
+int main(int argc, char* argv[])
+try
+{
+    if (readCommandLine(argc, argv))
+    {
+        if (setup)
+        {
+            if (!fs::is_directory("vcpkg"))
+            {
+                execute("git clone https://github.com/microsoft/vcpkg.git");
+                execute("git clone --progress -v vcpkg vcpkg_clone");
+                fs::current_path("vcpkg");
+                #ifdef WIN32
+                    execute(".\\bootstrap-vcpkg.bat -disableMetrics");
+                #else
+                    execute("./bootstrap-vcpkg.sh -disableMetrics");
+                #endif
+            }
+            else
+            {
+                fs::current_path("vcpkg");
+            }
+
+            if (fs::exists(sdkRootPath / "contrib" / "cmake" / "vcpkg_extra_triplets" / (triplet + ".cmake")))
```

Comment:
Minor but `std::filesystem::path::operator/` is implemented as a string concatenation and I'm pretty sure `'/'` is acceptable as a separator on any platform so we may want to just hardcode `"contrib/cmake/vcpkg_extra_triplets"` here and elsewhere

Reply:
From my understanding of the filesystem library, the separated out  / is the platform-independent way to code - some implementations may work that way, but we can't say they all do.   I just double checked at https://en.cppreference.com/w/cpp/filesystem, ```The native format (e.g. which characters are used as separators) and character encoding of the string representation of a path (the pathname) is implementation-defined, this library provides portable representation of paths.```